src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Casts DateTimeInterface related classes to array representation.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+class DateCaster
+{
+    public static function castDate(\DateTimeInterface $d, array $a, Stub $stub, $isNested, $filter)
+    {
+        $prefix = Caster::PREFIX_VIRTUAL;
+
+        $a = array();
+        $a[$prefix.'date'] = new ConstStub(
+            $d->format('Y-m-d H:i:s.u '.($d->getTimeZone()->getLocation() ? 'e (P)' : 'P')),
+            sprintf(
+                "literal: %s\nΔnow: %s",
+                $d->format('l, j F Y'),
+                (new \DateTime())->diff($d)->format('%R %yy %mm %dd %H:%I:%S')
```

Comment:
I'am rather favor of https://github.com/symfony/symfony/pull/22431#discussion_r114303969

Explanation: if it's 12:00 in Mali (+00:00) and 14:00 in Zurich (+02:00), it's the same moment in the **T**ime, there is no interval of time between. But if we set Zurich's tz, we will have 2h of interval:

```php
>>> (new \DateTime('12:00+00:00'))->diff(new \DateTime('14:00+02:00'))->format('%R %yy %mm %dd %H:%I:%S')
=> "+ 0y 0m 0d 00:00:00" // Ok

>>> (new \DateTime('12:00+02:00'))->diff(new \DateTime('14:00+02:00'))->format('%R %yy %mm %dd %H:%I:%S')
=> "+ 0y 0m 0d 02:00:00" // Ko
```